### PR TITLE
feat: added HTTP webhook validation to spin SDK

### PR DESCRIPTION
**Pull request by Mossaka**
this PR adds the http webhook validation function to Spin SDK; The detailed specification for validation is written [here](https://github.com/cloudevents/spec/blob/main/cloudevents/http-webhook.md). 

This is the first PR to implement the Spin SIP #398 . See related discussion here #394

### CODE REVIEW

File: sdk/rust/src/webhook.rs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+use crate::outbound_http::send_request;
+use anyhow::Result;
+use std::str::FromStr;
+
+static ORIGIN_REQUEST_HEADER: &str = "webhook-request-origin";
+static CALLBACK_REQUEST_HEADER: &str = "webhook-request-callback";
+static RATE_RESPONSE_HEADER: &str = "webhook-allowed-rate";
+static ORIGIN_RESPONSE_HEADER: &str = "webhook-allowed-origin";
+
+/// validate the webhook according to this [spec](https://github.com/cloudevents/spec/blob/v1.0/http-webhook.md#4-abuse-protection).
```
Comment (radu-matei) [inline sdk/rust/src/webhook.rs:11]: nit: could you please use capital letters for doc comments?

File: sdk/rust/src/webhook.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+pub fn validate_webhook(req: Request, callback_allowed: bool, rate: &str) -> Result<Response> {
+    if req.method() != http::Method::OPTIONS {
+        return Err(anyhow::anyhow!("invalid method"));
```
Comment (radu-matei) [inline sdk/rust/src/webhook.rs:14]: This can be simplified with `bail!("invalid method")`.

**Comment by lann** [comment]
We have been discussing package organization and I think it would make sense for this to live in an outside repository. @michelleN suggests that a new `deislabs/spin-cloudevents` repo might be a good home for this code plus any future cloudevents trigger executor(s) (which should become much easier with #513!).

**Comment by lann** [comment]
Oh! We were also discussing adding a "related projects" section to this repo's README, which would include a link to your repo.

**Comment by Mossaka** [comment]
@lann okay I will try #513 out on a new repo.

**Comment by lann** [comment]
Is this spec used anywhere apart from CloudEvents?

**Comment by Mossaka** [comment]
> Is this spec used anywhere apart from CloudEvents?

so far I haven't seen it being used in other places.

**Comment by radu-matei** [comment]
Thanks a lot for the contribution, @Mossaka!

It seems like it would make sense for this functionality to be part of a repo that was focused on adding CloudEvents support for Spin, rather than in the Spin repo itself.
As we were discussing in the developer meeting, that repo could evolve to include support for subscription managers and other things.

Then we can link the repo in the documentation and in the readme.
What do you think?

**Comment by Mossaka** [comment]
@radu-matei yep I will move to spin-cloudevents repo.

**Pull request closed without merging**